wrapanapi/systems/google.py

```diff
@@ -0,0 +1,798 @@
+        return self.get_template(name, self._project)
+
+    def list_bucket(self):
+        buckets = self._get_all_buckets()
+        return [bucket.get('name') for bucket in buckets.get('items', [])]
+
+    def list_forwarding_rules(self):
+        rules = self._get_all_forwarding_rules()
+        return [forwarding_rule.get('name') for forwarding_rule in rules]
+
+    def _find_forwarding_rule_by_name(self, forwarding_rule_name):
+        try:
+            forwarding_rule = self._forwarding_rules.get(
+                project=self._project, zone=self._zone,
+                forwardingRule=forwarding_rule_name).execute()
+            return forwarding_rule
+        except Exception:
+            raise NotFoundError
+
+    def _check_operation_result(self, result):
+        if result['status'] == 'DONE':
+            self.logger.info("The operation '%s' -> DONE", result['name'])
+            if 'error' in result:
+                self.logger.error("Error during operation '%s'", result['name'])
+                self.logger.error("Error details: %s", result['error'])
+                raise Exception(result['error'])
+            return True
+        return False
+
+    def is_global_operation_done(self, operation_name):
+        result = self._compute.globalOperations().get(
+            project=self._project,
+            operation=operation_name).execute()
+        self._check_operation_result(result)
+
+    def is_zone_operation_done(self, operation_name, zone=None):
+        if not zone:
+            zone = self._zone
+        result = self._compute.zoneOperations().get(
+            project=self._project,
+            zone=zone,
+            operation=operation_name).execute()
+        self._check_operation_result(result)
+
+    def create_bucket(self, bucket_name):
+        """ Create bucket
+        Args:
+            bucket_name: Unique name of bucket
+        """
+        if not self.bucket_exists(bucket_name):
+            self._buckets.insert(
+                project=self._project, body={"name": "{}".format(bucket_name)}).execute()
+            self.logger.info("Bucket '%s' was created", bucket_name)
+        else:
+            self.logger.info("Bucket '%s' was not created, exists already", bucket_name)
+
+    def create_image(self, image_name, bucket_url, timeout=360):
+        """ Create image from file
+        Args:
+            image_name: Unique name of image
+            bucket_url: url to image file in bucket
+            timeout: time to wait for operation
+        """
+        images = self._compute.images()
+        data = {
+            "name": image_name,
+            "rawDisk": {"source": bucket_url}
+        }
+        operation = images.insert(project=self._project, body=data).execute()
+        wait_for(
+            lambda: self.is_global_operation_done(operation['name']),
+            delay=0.5, num_sec=timeout, message=" Creating image {}".format(image_name)
+        )
+        return self.get_template(image_name)
```

Comment:
This whole method is a duplicate implementation of `create_template`, which is the TemplateMixin abstractmethod, with only this return being different in that it doesn't pass the project to get_template.

Why is this? Seems unnecessary to implement twice?

Reply:
Good catch! :) Probably just a mixup I missed with all the shuffling around and renaming I've been doing :) Removing this